### ISSUE TITLE
Make normalize_path configurable to allow for trailing slashes

### ISSUE THREAD

**Issue by alper** [tower-rs/tower-http#530]
I think paths should be normalised to end with a slash.

See how (and why) Django does this:

* https://stackoverflow.com/a/42213107/102315
* https://docs.djangoproject.com/en/dev/ref/settings/#std-setting-APPEND_SLASH

Since `tower-http/src/normalize_path.rs` has already gone in the other direction, I guess the best way forward would be to make it configurable so that those who prefer trailing slashes can get them?

I'll happily make the change.

**Comment by jplatte** [comment]
The existing constructor is named `trim_trailing_slash` (rather than `new`) to allow for other constructors / normalization styles to be added down the line. So feel free to send a PR. I think I had a short discussion with somebody around what to name the "add trailing slash if missing" style before, but I don't know where that was. If it's not in another issue or GH discussion, it's probably going to be impossible to find. Settling the exact name can happen on an otherwise finished PR, IMHO.

**Comment by daalfox** [comment]
@alper @jplatte Would you mind if I give this a try?

**Comment by jplatte** [comment]
Go ahead!

**Comment by alper** [comment]
Thanks for this!

**Comment by jplatte** [comment]
`normalize_path` was never meant to do redirects. If you're using axum, you can use [`route_with_tsr`](https://docs.rs/axum-extra/latest/axum_extra/routing/trait.RouterExt.html#tymethod.route_with_tsr) to get that.

**Comment by alper** [comment]
That seems to be then exactly what I'm after. But what is `normalize_path` for then?

**Comment by jplatte** [comment]
Yes, the effect of the layer will be observed by the tracing layer if that is applied after `NormalizePath` inside the router. However, the routing itself will happen before `NormalizePath` erases or adds trailing slashes, which I would expect to defeat the point of using it.
